### PR TITLE
Fix deprecation warnings for rails 4.2

### PR DESCRIPTION
**Pull request by arthurnn**
review @dylanahsmith @camilo 

still 2 failing tests. Will fix them next.

### CODE REVIEW

File: lib/identity_cache/query_api.rb [hunk 1]
```diff
@@ -80,7 +80,13 @@ def fetch_multi(*ids)
...
         if coder.present? && coder.has_key?(:class)
           record = coder[:class].allocate
-          unless coder[:class].serialized_attributes.empty?
+          empty_serialized_attrs =
+            if defined?(ActiveRecord::Type::Serialized)
+              coder[:class].columns.find { |t| t.cast_type.is_a?(ActiveRecord::Type::Serialized) }.nil?
+            else
+              coder[:class].serialized_attributes.empty?
+            end
+          unless empty_serialized_attrs
```
Comment (arthurnn) [inline lib/identity_cache/query_api.rb]: @dylanahsmith do you recall why we need this condition at all?

Comment (dylanahsmith) [inline lib/identity_cache/query_api.rb]: Nope, because I didn't add it.  https://github.com/Shopify/identity_cache/pull/88

Comment (arthurnn) [inline lib/identity_cache/query_api.rb]: cc @fbogsany

**Comment by fbogsany** [comment]
#88 for background info.

**Pull request merged**